lib/vector-core/src/event/metric/tags.rs

```diff
@@ -451,6 +451,10 @@ impl MetricTags {
             .and_then(TagValueSet::into_single)
     }
 
+    pub fn set_multi_value_tag(&mut self, name: String, values: Vec<TagValue>) {
+        self.0.insert(name, TagValueSet::from(values));
```

Comment:
Since `TagValueSet` doesn't reuse the incoming `Vec`, I would prefer a parameter type of something like `impl IntoIterator<Item = TagValue>` here, which can support a number of different container types. Since this type is `MetricTags`, I think the `_tag` suffix is redundant.